pints/_mcmc/__init__.py

```diff
@@ -1102,3 +1102,76 @@ def mcmc_sample(log_pdf, chains, x0, sigma0=None, method=None):
     """
     return MCMCController(    # pragma: no cover
         log_pdf, chains, x0, sigma0, method=method).run()
+
+
+def initialise_finite(log_pdf, chains, random_sampler=None, max_n_tries=50,
+                      parallel=False, n_workers=None):
+    """
+    Draws parameter values from a given sampling distribution until either
+    finite values for each of ``chains`` have been generated or the number of
+    attempts exceeds ``num_tries``.
+
+    If log_pdf is of :class:`LogPosterior`, then the
+    `log_pdf.log_prior().sample` method is used for initialisation, although
+    this is overruled by `random_sampler` if it is supplied.
+
+    Parameters
+    ----------
+    log_pdf : pints.LogPDF
+        A :class:`LogPDF` function that evaluates points in the parameter
+        space.
+    chains : int
+        The number of initial values to generate.
+    random_sampler : stochastic function
+        A function that when called returns draws from a probability
+        distribution of same dimensionality as `log_pdf`. The only argument
+        to this function should be an integer specifying the number of draws.
+    max_n_tries : int
+        Number of attempts to find a finite initial value for each of
+        ``chains``.
+    parallel : Boolean
+        Whether to evaluate log_pdf in parallel (defaults to False).
+    n_workers : int
+        Number of workers on which to run parallel evaluation.
+    """
+    if random_sampler is not None and not callable(random_sampler):
+        raise ValueError("random_sampler must be a callable function.")
+
+    if random_sampler is None:
+        if isinstance(log_pdf, pints.LogPosterior):
+            random_sampler = log_pdf.log_prior().sample
+        else:
+            raise ValueError("If log_pdf not of class pints.LogPosterior " +
+                             "then random_sampler must be supplied.")
+
+    if chains < 1:
+        raise ValueError("chains must be 1 or more.")
+
+    if parallel:
+        n_workers = min(pints.ParallelEvaluator.cpu_count(), chains)
+        evaluator = pints.ParallelEvaluator(log_pdf, n_workers=n_workers)
+    else:
+        evaluator = pints.SequentialEvaluator(log_pdf)
+
+    max_n_tries_chains = max_n_tries * chains
```

Comment:
I don't like the logic here! Max tries should either be max tries total, or max per chain.
You could do max per chain as:
```
for i in range(max_tries):
    if cool then break
```
You could do max total as:
```
tries_remaining = max_tries
chains_remaining = chains
while tries_remaining >= chains_remaining:
  tries_remaining -= chains_remaining
  ...
if chains_remaining:
  raise Error('boooo')
```